core/src/com/unciv/ui/screens/diplomacyscreen/OffersListScroll.kt

```diff
@@ -109,7 +112,9 @@ class OffersListScroll(
                         else -> 1
                     }
 
-                if (offer.isTradable() && offer.name != Constants.peaceTreaty) { // can't disable peace treaty!
+                if (offer.isTradable() && offer.name != Constants.peaceTreaty // can't disable peace treaty!
+                    && (offer.name != Constants.researchAgreement // If we have a research agreement make sure the total gold of both Civs is higher than the total cost
+                        || (ourCiv.gold + theirCiv.gold > ourCiv.diplomacyFunctions.getResearchAgreementCost() * 2))) { 
```

Comment:
Seems to me this is the wrong place to make this check anyway - we should be making it when populating the offersToDisplay, NOT filtering it out later. That way, we won't have to send the 'ourCiv' and 'theirCiv' parameters to this function at all

Reply:
The only filtering that is being done to the research agreement is at tradeLogic.getAvailableOffers(). In OffersListScroll we are only disabling and enabling it based on if both Civs have enough combined gold or not. If they don't have enough gold we should still display the research agreement button but disable it. The only place to disable a tradeButton is here in offersListScroll.
In Civ V the player can see the research pact and it's cost even if it is not affordable.

I do agree that passing ourCiv and theirCiv is a bit much though.